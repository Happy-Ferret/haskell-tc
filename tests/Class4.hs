module Class4 where

data String

class Show a where
  show :: a -> String

fn1 x y = (fn2 x, y)
  where
    fn2 x = show x
