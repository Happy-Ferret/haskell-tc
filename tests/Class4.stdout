Bindings:
:: âˆ€ v10 v9. .Show v10 â‡’ v10 â†’  v9 â†’  (String
                                      ,v9)
  [4mfn1 x y = (fn2 x, y)[0m
  [4m  where[0m
  [4m    fn2 x = show x[0m
:: v10
  fn1 [4mx[0m y = (fn2 x, y)
:: v9
  fn1 x [4my[0m = (fn2 x, y)
:: âˆ€ v7. .Show v7 â‡’ v7 â†’  String
      [4mfn2 x = show x[0m
:: v7
      fn2 [4mx[0m = show x

Proofs:
coercion: (âˆ€ v7. .Show v7 â‡’ v7 â†’  String) @ v10
  fn1 x y = ([4mfn2[0m x, y)
coercion: (âˆ€ a. .Show a â‡’ a â†’  String) @ v7
      fn2 x = [4mshow[0m x
